tests/attribute_tests.cpp

```diff
@@ -0,0 +1,58 @@
+#define BOOST_TEST_MODULE RadProtoAttributeTests
+
+#include "radproto/attribute.h"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#pragma GCC diagnostic ignored "-Wsign-compare"
+#pragma GCC diagnostic ignored "-Wparentheses"
+#include <boost/test/unit_test.hpp>
+#pragma GCC diagnostic pop
+
+BOOST_AUTO_TEST_SUITE(RadProtoAttributeTests)
+
+BOOST_AUTO_TEST_CASE(ClassStringDataConstructor)
+{
+    std::vector<uint8_t> d {'t', 'e', 's', 't'};
+    RadProto::String s(1, d.data(), d.size());
+
+    BOOST_CHECK_EQUAL(s.toString(), "test");
+
+    std::vector<uint8_t> values = s.toVector({}, {});
+    std::vector<uint8_t> expected {1, 6, 't', 'e', 's', 't'};
+
+    BOOST_CHECK_EQUAL_COLLECTIONS(values.begin(), values.end(), expected.begin(), expected.end());
+
+    BOOST_CHECK_EQUAL(s.type(), 1);
+}
+
+BOOST_AUTO_TEST_CASE(ClassStringValueConstructor)
+{
+    RadProto::String v(1, "test");
+
+    BOOST_CHECK_EQUAL(v.toString(), "test");
+
+    std::vector<uint8_t> values = v.toVector({}, {});
+    std::vector<uint8_t> expected {1, 6, 't', 'e', 's', 't'};
+
+    BOOST_CHECK_EQUAL_COLLECTIONS(values.begin(), values.end(), expected.begin(), expected.end());
+
+    BOOST_CHECK_EQUAL(v.type(), 1);
+}
+
+BOOST_AUTO_TEST_CASE(ClassStringClone)
+{
+    RadProto::String c(1, "test");
+
+    BOOST_CHECK_EQUAL(c.clone()->toString(), "test");
```

Comment:
Although it is not a big problem here, remember that `clone()` makes a copy in heap. And you need to delete it manually.

1. Do not call `clone()` in test statements, make a single clone and use it.
2. Take care of the resource, delete it after use.

I propose something like this:
```c++
RadProto::String s(1, "test");
std::unique_ptr<RadProto::Attribute> cs(c.clone());

BOOST_CHECK_EQUAL(cs->toString(), "test");
// ...
```